bladex/shaft.py

```
@@ -27,10 +28,14 @@ def generate_solid(self):
         :return: solid shaft
         :rtype: OCC.Core.TopoDS.TopoDS_Solid
         """
-        iges_reader = IGESControl_Reader()
-        iges_reader.ReadFile(self.filename)
-        iges_reader.TransferRoots()
-        shaft_compound = iges_reader.Shape()
+        ext = os.path.splitext(self.filename)[1][1:]
+        if ext=='stl':
+        	shaft_compound = read_stl_file(self.filename)
+        elif ext=='iges':
+        	iges_reader = IGESControl_Reader()
+        	iges_reader.ReadFile(self.filename)
+        	iges_reader.TransferRoots()
+        	shaft_compound = iges_reader.Shape()
```

please raise an error for different extensions